src/app/components/chat/client.ts

```diff
@@ -98,121 +92,50 @@ export class ChatClient {
-
 	if (user.value) {
-		const channel = await createChatUserChannel(chat, { userId: user.value.id });
+		const channel = createChatUserChannel(chat, { userId: user.value.id });
+		await channel.joinPromise;
 
 		chat.userChannel = channel;
 		chat.populated = true;
 	}
 }
 
-async function joinRoomChannel(chat: ChatClient, roomId: number) {
-	if (chat.pollingRoomId === roomId) {
-		chat.logger.info('Do not attempt to join the same room twice.', roomId);
-		return;
-	}
-
-	return await createChatRoomChannel(chat, { roomId, instanced: false });
-}
-
 /**
- * Called by the chat room channel to set itself up fully after joining.
+ * Call this to open a chat room window. This will trigger the Shell to show the
+ * chat window which will do the actual connection.
  */
-export function setupChatRoom(chat: ChatClient, room: ChatRoom, messages: ChatMessage[]) {
-	if (isRoomInstanced(chat, room.id) || !isInChatRoom(chat, room.id)) {
-		// Only set the room as "the" active room when it's not instanced.
-		if (!isRoomInstanced(chat, room.id)) {
-			setChatRoom(chat, room);
-		}
-
-		// Clear out any old messages so we don't use old data from the model
-		// store.
-		room.messages = [];
-		processNewChatOutput(room, messages, true);
-	}
-}
+export function openChatRoom(chat: ChatClient, roomId: number) {
+	const { openChatPane } = chat.appStore;
 
-export function setChatRoom(chat: ChatClient, newRoom: ChatRoom | undefined) {
-	leaveChatRoom(chat);
+	chat.activeRoomId = roomId;
+	chat.setSessionRoomId(roomId);
 
-	if (newRoom) {
-		if (chat.currentUser && chat.isFocused) {
-			chat.roomChannels[newRoom.id].pushFocus();
-		}
-
-		chat.sessionRoomId = newRoom.id;
-	}
-
-	chat.room = newRoom || null;
+	openChatPane();
 }
 
 /**
- * Call this to open a room. It'll do the correct thing to either open the chat
- * if closed, or enter the room.
+ * Call this to close the chat room window. When the window tears down, the
+ * disconnection from the room will happen.
  */
-export function enterChatRoom(chat: ChatClient, roomId: number) {
-	if (isInChatRoom(chat, roomId)) {
-		return;
-	}
-
-	const { visibleLeftPane, toggleChatPane } = chat.appStore;
-
-	// If the chat isn't visible yet, set the session room to this new room and
-	// open it. That will in turn do the entry. Otherwise we want to just switch
-	// rooms.
-	if (visibleLeftPane.value !== 'chat') {
-		chat.sessionRoomId = roomId;
-		toggleChatPane();
-	} else {
-		joinRoomChannel(chat, roomId);
-	}
-}
-
-export function leaveChatRoom(chat: ChatClient, room: ChatRoom | null = null) {
-	if (chat.messageEditing) {
-		setMessageEditing(chat, null);
-	}
-
-	if (room === null) {
-		room = chat.room;
-	}
-
-	if (!room) {
-		return;
-	}
-
-	const channel = chat.roomChannels[room.id];
-	if (channel) {
-		stopTyping(chat, room);
-		delete chat.roomChannels[room.id];
-		channel.channelController.leave();
-		chat.pollingRoomId = -1;
-	}
+export function closeChatRoom(chat: ChatClient) {
+	chat.activeRoomId = null;
```

Comment:
Does this still close the chat pane like the leaveChatRoom function did?

Reply:
Nope. I realized that in all cases (except one) we didn't want to close the sidebar after leaving the room. So I changed this function and then updated that one place to also close the chat pane after closing the chat room.